### PR TITLE
fix: [CDS-83763]: support input value when sent on whenCondition

### PR DESCRIPTION
**Pull request by lasdox**
## Describe your changes
The changes on this PR are fixing and defining some rules for when.condition node.

- if when.condition is null, we consider the default as to always execute
- if when.condition reaches backend with value <+input>, we consider it as false

## Test list
- [x] Sending null when.condition should evaluate to true
- [x] Sending <+input> as value should evaluate to false
- [x] Sending <+input> with default value should evaluate to default value  

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55192)

### CODE REVIEW

**Comment by lasdox** [comment]
trigger uts

**Comment by lasdox** [comment]
trigger uts

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55192)* status: 2 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/facilitation/RunPreFacilitationChecker.java` line 54 at r2](https://reviewable.io/reviews/harness/harness-core/55192#-NimPGeT9g2nZAjZIZVl:-NimPGeT9g2nZAjZIZVm:bo2ursz) ([raw file](https://github.com/harness/harness-core/blob/b93a7044ff23d83a9df7939c588eed0c5de688d0/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/facilitation/RunPreFacilitationChecker.java#L54)):*
> ```Java
>   private static final String MAX_LEVELS_LIMIT_REACHED_ERROR =
>       "The pipeline has reached the maximum nesting allowed for an execution. Please simplify the pipeline configuration so that it does not breach the allowed limit of nesting";
>   private static final Map<String, Object> CHECKER_EXPRESSION_CTX = Map.of("input", true);
> ```

Instead of this, we can add handling in RunInfoUtils which is used by all planCreators to combine when condition.
If condition is <+input> then don't consider would be apt solution

**Comment by lasdox** [comment]
trigger uts

File: clients/pipeline-service/pms-sdk-core/src/main/java/io/harness/when/utils/RunInfoUtils.java [hunk 1]
```diff
@@ -35,6 +35,7 @@ public class RunInfoUtils {
   String PIPELINE_FAILURE = "OnPipelineFailure";
   String ALWAYS = "Always";
   String ROLLBACK_MODE_EXECUTION = "OnRollbackModeExecution";
+  private static String RUNTIME_EXSPRESSION = "<+input>";
```
Comment (ankittiwari-harness) [inline clients/pipeline-service/pms-sdk-core/src/main/java/io/harness/when/utils/RunInfoUtils.java:38]: nit: expression typo

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/facilitation/RunPreFacilitationChecker.java` line 54 at r2](https://reviewable.io/reviews/harness/harness-core/55192#-NimPGeT9g2nZAjZIZVl:-Nio6l2B0pTqDgzy1m1Q:b32v57) ([raw file](https://github.com/harness/harness-core/blob/b93a7044ff23d83a9df7939c588eed0c5de688d0/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/facilitation/RunPreFacilitationChecker.java#L54)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Instead of this, we can add handling in RunInfoUtils which is used by all planCreators to combine when condition.
If condition is <+input> then don't consider would be apt solution
</blockquote></details>

+1

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55192)* status: 4 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`clients/pipeline-service/pms-sdk-core/src/main/java/io/harness/when/utils/RunInfoUtils.java` line 38 at r3](https://reviewable.io/reviews/harness/harness-core/55192#gh-1387947816:-Nio5ucp5oXe4Wjco24M:bbrezr8) ([raw file](https://github.com/harness/harness-core/blob/86b882b19c49cf4ef208aaac66392f1d50925714/clients/pipeline-service/pms-sdk-core/src/main/java/io/harness/when/utils/RunInfoUtils.java#L38)):*
<details><summary><i>Previously, ankittiwari-harness (Ankit Tiwari) wrote…</i></summary><blockquote>

nit: expression typo
</blockquote></details>

we already have this via NGExpressionUtils.matchesInputSetPattern
___
*[`clients/pipeline-service/pms-sdk-core/src/main/java/io/harness/when/utils/RunInfoUtils.java` line 106 at r3](https://reviewable.io/reviews/harness/harness-core/55192#-Nio6KdCECTRul0HOPsm:-Nio6KdCECTRul0HOPsn:bs8vh6a) ([raw file](https://github.com/harness/harness-core/blob/86b882b19c49cf4ef208aaac66392f1d50925714/clients/pipeline-service/pms-sdk-core/src/main/java/io/harness/when/utils/RunInfoUtils.java#L106)):*
> ```Java
> 
>   private String combineExpressions(String statusExpression, String conditionExpression) {
>     if (EmptyPredicate.isNotEmpty(conditionExpression) && !RUNTIME_EXSPRESSION.equals(conditionExpression)) {
> ```

we discussed solution with product

in this method -> getGivenRunCondition should change that if empty string then return as it is and test it should work as false
cc: @sahilhindwani-harness

**Comment by lasdox** [comment]
trigger uts

**Review by archit-harness — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55192)* status: 4 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`clients/pipeline-service/pms-sdk-core/src/main/java/io/harness/when/utils/RunInfoUtils.java` line 106 at r3](https://reviewable.io/reviews/harness/harness-core/55192#-Nio6KdCECTRul0HOPsm:-NioYCXe9tYCfUcDodZf:b-683c3n) ([raw file](https://github.com/harness/harness-core/blob/86b882b19c49cf4ef208aaac66392f1d50925714/clients/pipeline-service/pms-sdk-core/src/main/java/io/harness/when/utils/RunInfoUtils.java#L106)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

we discussed solution with product

in this method -> getGivenRunCondition should change that if empty string then return as it is and test it should work as false
cc: @sahilhindwani-harness
</blockquote></details>

done @sahilhindwani-harness resolving as i will be OOO

**Comment by lasdox** [comment]
trigger uts

**Comment by lasdox** [comment]
trigger messagecheck

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 2 of 4 files at r3, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55192)* status: Some of the checks are too old. 4 file(s) to review. 1 change requests. 1 of 2 approvals obtained.

**Comment by lasdox** [comment]
trigger review

**Pull request merged**